.github/workflows/main.yml

```diff
@@ -0,0 +1,26 @@
+name: SpackBot CI
+
+on:
+  pull_request: []
+
+jobs:
+  formatting:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Check formatting
+        run: |
+          export PATH="/usr/share/miniconda/bin:$PATH"
+          pip install black
+          black --check --diff spackbot
+
+      # Likely only maintainers will work on Spackbot
+      - name: Comment Pull Request
+        if: github.event_name == 'pull_request' && failure()
+        uses: marocchino/sticky-pull-request-comment@v1.1.0
+        with:
+          message: 'Please format your code with [black](https://black.readthedocs.io): `black spackbot`.'
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          
+      # TODO add tests here?
```

Comment:
Can you please add `flake8` here, with config that matches the black config? See https://black.readthedocs.io/en/stable/guides/using_black_with_other_tools.html?highlight=flake8#flake8

`flake8` will catch the stuff that `black` can't fix -- e.g. there are some lines in this PR that wouldn't pass `flake8` but `black` currently doesn't break up long strings.

Reply:
I think this is out of scope for this PR, probably adding this file is too (that's my fault).

Reply:
ok sure -- this can be separate